net/nimble/host/src/ble_gap.c

```diff
@@ -1803,6 +1855,64 @@ ble_gap_adv_params_tx(uint8_t own_addr_type, const ble_addr_t *peer_addr,
                       const struct ble_gap_adv_params *adv_params)
 
 {
+#if MYNEWT_VAL(BLE_EXT_ADV)
+    struct hci_ext_adv_params hci_adv_params;
+    uint8_t buf[BLE_HCI_CMD_HDR_LEN + BLE_HCI_LE_SET_EXT_ADV_PARAM_LEN];
+    uint16_t min_int = 0, max_int = 0;
+    int rc;
+
+    if (peer_addr == NULL) {
+        peer_addr = BLE_ADDR_ANY;
+    }
+
+    hci_adv_params.own_addr_type = own_addr_type;
+    hci_adv_params.peer_addr_type = peer_addr->type;
+    memcpy(hci_adv_params.peer_addr, peer_addr->val,
+           sizeof hci_adv_params.peer_addr);
+
+    /* Fill optional fields if application did not specify them. */
+    if (adv_params->itvl_min == 0 && adv_params->itvl_max == 0) {
+        ble_gap_adv_dflt_itvls(adv_params->conn_mode, &min_int, &max_int);
+
+        /* TODO for now limited to legacy values*/
+        hci_adv_params.min_interval = min_int;
+        hci_adv_params.max_interval = max_int;
+
+    } else {
+        hci_adv_params.min_interval = adv_params->itvl_min;
+        hci_adv_params.max_interval = adv_params->itvl_max;
+    }
+    if (adv_params->channel_map == 0) {
+        hci_adv_params.chan_map = BLE_GAP_ADV_DFLT_CHANNEL_MAP;
+    } else {
+        hci_adv_params.chan_map = adv_params->channel_map;
+    }
+
+    /* Zero is the default value for filter policy and high duty cycle */
+    hci_adv_params.filter_policy = adv_params->filter_policy;
+    hci_adv_params.tx_power = ext_adv_tx_pwr;
+
+    /* if phy was not set this means legacy advertising PDUs */
+    if (ext_adv_pri_phy == 0) {
+        hci_adv_params.properties = ble_gap_adv_type(adv_params);
```

Comment:
I don't think this does what it should: ble_gap_adv_type() returns ordinal value 0..4 while properties for legacy pdus shall be one of bitmasks defined in spec